### PR TITLE
feature: take snapshot for grafana monitor

### PR DESCRIPTION
**Pull request by kongove**
Grafana snapshot gives us quick and visual impression,
it's not convenient to load prometheus data from test
result. This patch supports to take monitor snapshot
at the end of test.

### CODE REVIEW

File: sdcm/cluster.py [hunk 1]
```diff
@@ -1792,6 +1798,23 @@ def node_setup(node):
...
+                            verbose=True)
+                process.run("cd phantomjs-2.1.1-linux-x86_64  && "
+                            "sed -e 's/}, 200);/}, 10000);/' examples/rasterize.js > rasterize.js && "
+                            "bin/phantomjs rasterize.js \"http://%s:3000/dashboard/db/scylla-per-server-metrics-1-6?from=%s&to=now\" \"%s/grafana-snapshot-%s.png\" && "
+                            "cd -" % (node.public_ip_address,
+                                      self.grafana_start_time,
+                                      self.logdir, n), shell=True, verbose=True)
+            except Exception, details:
+                self.log.error('Error taking monitor snapshot: %s',
+                               str(details))
```
Comment (lmr) [inline sdcm/cluster.py]: Nice, it's an interesting solution for taking a screenshot of the data. I'll keep an eye into phantomjs.

Comment (lmr) [inline sdcm/cluster.py]: I don't think it would be necessary to `cd -`, since each command is executed on its own bash instance.

Comment (lmr) [inline sdcm/cluster.py]: I tend to like splitting up commands in easier to understand chunks. So for what is worth, what you did is fine, but consider splitting up for added clarity.

File: sdcm/cluster.py [hunk 2]
```diff
@@ -1792,6 +1798,23 @@ def node_setup(node):
         time_elapsed = time.time() - start_time
         self.log.debug('Setup duration -> %s s', int(time_elapsed))
 
+    def get_monitor_snapshot(self):
+        for n, node in enumerate(self.nodes):
+            try:
+                process.run('wget "https://bitbucket.org/ariya/phantomjs/downloads/phantomjs-2.1.1-linux-x86_64.tar.bz2"',
+                            verbose=True)
+                process.run('tar xvfj phantomjs-2.1.1-linux-x86_64.tar.bz2',
```
Comment (lmr) [inline sdcm/cluster.py]: Maybe we could reduce verbosity in all uncompressing commands, as the verbose output doesn't help a whole lot. This is just me thinking out loud, no action would be necessary here.

Comment (kongove) [inline sdcm/cluster.py]: Agree, I will clean it.

File: sdcm/cluster.py (hunk 1 above)
Comment (kongove) [inline sdcm/cluster.py]: I will remove 'cd -', and try to use variables to save url & path.

File: sdcm/cluster.py [hunk 3]
```diff
@@ -1769,6 +1771,10 @@ def node_setup(node):
             node.setup_prometheus(targets=targets)
             node.install_grafana()
             node.setup_grafana()
+            # The time will be used in url of Grafana monitor,
+            # the data from this point to the end of test will
+            # be captured.
+            self.grafana_start_time = str(time.time()).split('.')[0] + '000'
```
Comment (kongove) [inline sdcm/cluster.py]: Let's save time.time() to `grafana_start_time`, not a non-standard string. And convert to timestamp string insider get_monitor_snapshot()

Comment (kongove) [inline sdcm/cluster.py]: Let's save time.time() to `grafana_start_time`, not a non-standard string. And convert to timestamp string insider get_monitor_snapshot()

File: sdcm/cluster.py [hunk 4]
```diff
@@ -1792,6 +1798,23 @@ def node_setup(node):
         time_elapsed = time.time() - start_time
         self.log.debug('Setup duration -> %s s', int(time_elapsed))
 
+    def get_monitor_snapshot(self):
+        for n, node in enumerate(self.nodes):
+            try:
+                process.run('wget "https://bitbucket.org/ariya/phantomjs/downloads/phantomjs-2.1.1-linux-x86_64.tar.bz2"',
```
Comment (kongove) [inline sdcm/cluster.py]: Let's move wget out of for loop to avoid repeated downloading.

Comment (lmr) [inline sdcm/cluster.py]: Yep, I haven't noticed that.

File: sdcm/cluster.py (hunk 3 above)
Comment (lmr) [inline sdcm/cluster.py]: Right, that's a good idea too.

**Comment by kongove** [comment]
I just updated the patch, it split cmds and improve phantomjs setup.

**Pull request merged**